torchtext/prototype/models/t5/model.py

```diff
@@ -191,6 +202,12 @@ def forward(
             decoder_output = self.dropout4(decoder_output)
             decoder_hidden_states = decoder_hidden_states + (decoder_output,)
 
+            if self.linear_head:
+                if self.tie_word_embeddings:
+                    decoder_output = decoder_output * (self.config.embedding_dim ** -0.5)
```

Comment:
Can we mention what the purpose of `tie_word_embeddings` is and leave a link to Google's/HF's implementation? At a cursory glance, it's not obvious why we multiply the decoder output with `(self.config.embedding_dim ** -0.5)`

Reply:
So it looks like `tie_word_embeddings` is just a flag to indicate whether or not the encoder and decoder use the same embedding weights. It looks like this configuration exists in HF across a wider set of models, but for T5 it is defaulted to True. Likewise, our implementation has it so that the encoder and decoder always share the same embeddings. I'm guessing that the reason why the rescaling is dependent on this flag being True is because it rescales the decoder output by the embedding dimension, which I guess only works if the encoder and decoder share the same embedding size?

Given that our T5 model will always share these embeddings, I'm wondering if it makes sense to keep the rescaling, but get rid of the flag, and just include a code pointer to where HF is doing a similar rescaling.

Reply:
As per our live conversation, I will remove `tie_word_embeddings` but keep the re-scaling.